uber/site_sections/preregistration.py

```diff
@@ -198,12 +198,15 @@ def prereg_payment(self, session, payment_id, stripeToken):
         else:
             message = charge.charge_cc(stripeToken)
 
-        if message:
+        if message and charge.total_cost:
```

Comment:
This clashes with line 194, above - you've made sure that that message will literally never show. Let's take this out and figure out how to make it so they don't have to put their CC info in in the first place for promo codes.